libraries/vault_installation_binary.rb

```diff
@@ -46,8 +46,14 @@ def action_create
           basename: options[:archive_basename])
 
         notifying_block do
-          directory ::File.join(options[:extract_to], new_resource.version) do
-            recursive true
+          [
```

Comment:
Why do you need to create the directory since you added the `recursive true` property?